src/Language/Haskell/Exts/InternalParser.ly

```diff
@@ -2120,15 +2127,24 @@ Pattern Synonyms
 -----------------------------------------------------------------------------
 Deriving strategies
 
-> deriv_strategy :: { Maybe (DerivStrategy L) }
+> deriv_strategy_no_via :: { DerivStrategy L }
 >       : 'stock'               {% do { checkEnabled DerivingStrategies
->                                     ; return (Just (DerivStock (nIS $1))) } }
+>                                     ; return (DerivStock (nIS $1)) } }
 >       | 'anyclass'            {% do { checkEnabled DerivingStrategies
 >                                     ; checkEnabled DeriveAnyClass
->                                     ; return (Just (DerivAnyclass (nIS $1))) } }
+>                                     ; return (DerivAnyclass (nIS $1)) } }
 >       | 'newtype'             {% do { checkEnabled DerivingStrategies
 >                                     ; checkEnabled GeneralizedNewtypeDeriving
->                                     ; return (Just (DerivNewtype (nIS $1))) } }
+>                                     ; return (DerivNewtype (nIS $1)) } }
+
+> deriv_strategy_via :: { DerivStrategy L }
+>       : 'via' truedtype       {% do { checkEnabled DerivingVia
+>                                     ; checkEnabled DerivingStrategies
```

Comment:
Why two checks? Does one not imply the other anyway?

Reply:
Indeed, that's a good point. I suppose checking for `DerivingVia` would be sufficient, then.